Move to QProcesses with custom ProcessManager

julianbreiteneicher opened the pull request:
Instead of `QThreads` running plumbum commands, this PR introduces `QProcesses` which are used for long running commands (like cloning git repos or building VaRA).
In contrast to `QThreads`, `QProcesses` can be terminated reliably.
This is especially important, if the user closes the application during such a long running task.
In this case, the processes are terminated correctly and the application doesn't hang any more.
This is implemented with a custom `ProcessManager` class with manages the processes.

julianbreiteneicher committed:
Add first (still buggy) version of QProcess migration

julianbreiteneicher committed:
Fix 'already shutdown' bug in ProcessManager

julianbreiteneicher committed:
Remove debugging output

julianbreiteneicher committed:
Add correct handling for download termination by user

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Correctly terminate application if build process is killed

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Add missing exceptions file

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Add contextmanager to easily create and start processes

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Fix small formatting issue

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Merge branch 'vara-dev' into f-ProcessPool

julianbreiteneicher committed:
Fix merge mistake

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Remove obsolete print statement

If the cmake process is interrupted by the user, we delete
the build dir. However, this does not consistently work,
so remove the text that indicates otherwise.
We still try to delete is, but it doesn't always work.

julianbreiteneicher committed:
Fix crash when the git fetch process was interrupted

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- varats/vara_manager.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Fix missing workdir for ninja process

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- varats/vara_manager.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Remove unused imports

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- varats/vara_manager.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

julianbreiteneicher committed:
Rename 'run_with_output' method

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- varats/vara_manager.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)

vulder committed:
Fixed formatting

vulder commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- varats/vara_manager.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3607370&bid=12791464)
